### PR TITLE
add REST API partial support section for Headless VM

### PR DESCRIPTION
**Pull request by sgratch**
I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md): @sgratch

This pull request needs review by: @jelkosz

### CODE REVIEW

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 1]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
 
 No changes in the DB
 
+#### REST API
+
+There is currently a  partial support for Headless in OVirt APIs.
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: a partial -> partial

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 2]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+The only option supported is editing an existing VM or Template or Instance-Type and change it via Ovirt APIs to Headless/non-Headless.
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: APIs -> API

Not sure what 'non-Headless' means.

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 3]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+This is done by using REST API with the "vms/\<vm-id\>/graphicsconsoles" extension.
+##### Usage:
+1. In case an existing VM or Template or Instance-Type is non-Headless and you want to change it to Headless, 
+    you just need to remove all graphic devices for this entity by calling the following API:
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: remove 'just'

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 4]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+
+    or
+
+    DELETE  ..api/templates/\<template-id\>/graphicsconsoles/\<console-id\>
+
+    or
+
+    DELETE  ..api/vms/\<vm-id\>/graphicsconsoles/\<console-id\>
+
+    while \<console-id\> is the ID of the graphic console you want to remove (spice or vnc).
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: graphic -> graphics
want -> wish
spice -> Spice
vnc -> VNC

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 5]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+  
+    When last graphic console is removed - the VM becomes Headless
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: graphic -> graphics

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 6]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+
+2. In case an existing VM or Template or Instance-Type is Headless and you want to change it to non-Headless,
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: want -> wish

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 7]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+    or
+
+    DELETE  ..api/vms/\<vm-id\>/graphicsconsoles/\<console-id\>
+
+    while \<console-id\> is the ID of the graphic console you want to remove (spice or vnc).
+  
+    When last graphic console is removed - the VM becomes Headless
+
+2. In case an existing VM or Template or Instance-Type is Headless and you want to change it to non-Headless, 
+    you just need to add a graphic device for this entity by calling the following APIs:
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: remove 'just'
graphic -> graphics
'graphics device', or 'graphics console' device?

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 8]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+    
+    POST  ..api/vms/\<vm-id\>/graphicsconsoles
+    
+    \<graphics_console\>
+    \<protocol\>type\</protocol\>
+    \</graphics_console\>
+    
+    while "type" can be "spice" or "vnc".
+    
+ 3. For creating a new Headless VM, you need to create a VM by the APIs (see REST API documentation for that). The VM will be created as non-Headless by default. Then use action 1 described above to change the VM from non-Headless to Headless
```
Comment (mykaul) [inline source/develop/release-management/features/virt/headless-vm.html.md]: by the APIs -> via the API

Perhaps a link to REST API documentation?

File: source/develop/release-management/features/virt/headless-vm.html.md (hunk 1 above)
Comment (jelkosz) [inline source/develop/release-management/features/virt/headless-vm.html.md]: - there are two spaces after the "currently a"
- for headless VMs in oVirt
- the oVirt is written with small o :)

File: source/develop/release-management/features/virt/headless-vm.html.md (hunk 2 above)
Comment (jelkosz) [inline source/develop/release-management/features/virt/headless-vm.html.md]: oVirt :)

File: source/develop/release-management/features/virt/headless-vm.html.md (hunk 5 above)
Comment (jelkosz) [inline source/develop/release-management/features/virt/headless-vm.html.md]: when the last

File: source/develop/release-management/features/virt/headless-vm.html.md [hunk 9]
```diff
@@ -86,11 +86,65 @@ In case of Headless VM / Template / Pool / Instance Type:
...
+    
+    
+
 ### Status
 
 *   Target Release: Ovirt 4.1
-*   Status: All is merged except REST API support for this feature (Backward compatibility is fully supported).
-*   REST API support Target Release: Ovirt 4.2
+*   Status: All is merged (Backward compatibility is fully supported).
+*   REST API full support Target Release: Ovirt 4.2
```
Comment (jelkosz) [inline source/develop/release-management/features/virt/headless-vm.html.md]: maybe a BZ URL would be good here.

**Comment by sgratch** [comment]
@mykaul @jelkosz - all suggested changes were done

**Comment by sgratch** [comment]
Added a third limitation regarding using REST API for Headless.

**Pull request merged**